### PR TITLE
I94 firebase emulator for testing

### LINKED ISSUE #94: Firebase Emulator for testing
## Basic Information

Init firebase emulator for auth, firestore and more

- [x] Init emulator
- [x] enable functions for email testing using extension

## Other Information
- Requires java11 or above to run emulator

Potential scopes:

### PR DESCRIPTION
**Pull request by NicholasChoong**
## Change Summary

Functions are not needed anymore but ill just leave it there

- [x] Init emulator
- [x] Added billing killswitch
- [x] Added pubsub test function to trigger the billing killswitch
- [x] Added helloWorld function (test)
- [x] Edited `firebase/init.ts` to connect to the emulator
- [x] Enabled offline support for firestore (tested)
- [x] Added firestore security rules
- [x] Removed the word `React`
- [x] Changed next config file to support ECMAScript modules
- [x] Temp fixed for next pwa throwing invalid next config waning message

- Most of changes are in the firebase/ and src/component/Firebase/unit.ts 

- As for the rest, just some code splitting for auth and firestore context and that changed the import path for the other components using the context. 

## Other Information
- [Node.js](https://nodejs.org/) version 8.0 or higher, but not higher than 16
- [Java](https://openjdk.java.net/install/) JDK version 11 or higher
- Install [Firebase CLI](https://firebase.google.com/docs/cli#windows-npm)

# Related Issue

- Resolve #94

### CODE REVIEW

File: src/components/PrivateRoute/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
```
Comment (skyheat) [inline src/components/PrivateRoute/index.tsx:1]: Not necessary as Next.js automatically imports react

File: src/context/Firebase/Auth/index.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { createContext, useContext, useEffect, useState } from 'react'
+import React, { useEffect, useState } from 'react'
```
Comment (skyheat) [inline src/context/Firebase/Auth/index.tsx:1]: Not necessary as Next.js automatically imports react

**Review by skyheat — CHANGES_REQUESTED** [review]
Looks great! Good job!

File: src/components/PrivateRoute/index.tsx (hunk 1 above)
Comment (NicholasChoong) [reply src/components/PrivateRoute/index.tsx:1]: Nice

**Review by skyheat — APPROVED** [review]
nice

**Pull request merged**